src/System.Security.Cryptography.Xml/tests/KeyInfoTest.cs

```diff
@@ -195,18 +196,139 @@ public void Complex()
+                        <Q>{dsaQ}</Q>
+                        <G>{dsaG}</G>
+                        <Y>{dsaY}</Y>
+                        <J>{dsaJ}</J>
+                        <Seed>{dsaSeed}</Seed>
+                        <PgenCounter>{dsaPgenCounter}</PgenCounter>
+                    </DSAKeyValue>
+                </KeyValue>
+                <KeyValue xmlns=""http://www.w3.org/2000/09/xmldsig#"">
+                    <RSAKeyValue>
+                        <Modulus>{rsaModulus}</Modulus>
+                        <Exponent>{rsaExponent}</Exponent>
+                    </RSAKeyValue>
+                </KeyValue>
+                <RetrievalElement URI=""{retrievalElementUri}"" />
+                <X509Data xmlns=""http://www.w3.org/2000/09/xmldsig#"">
+                    <X509Certificate>{x509cert}</X509Certificate>
+                </X509Data>
+            </KeyInfo>";
+
             XmlDocument doc = new XmlDocument();
             doc.LoadXml(value);
             info.LoadXml(doc.DocumentElement);
 
-            AssertCrypto.AssertXmlEquals("Import", value, (info.GetXml().OuterXml));
             Assert.Equal(5, info.Count);
+            int i = 0;
+            int pathsCovered = 0;
+            foreach (var clause in info)
+            {
+                i++;
+                
+                if (clause is KeyInfoName)
+                {
+                    pathsCovered |= 1 << 0;
+
+                    var name = clause as KeyInfoName;
+                    Assert.Equal(keyName, name.Value);
+                }
+                else if (clause is DSAKeyValue)
+                {
+                    pathsCovered |= 1 << 1;
+
+                    var dsaKV = clause as DSAKeyValue;
+                    DSA dsaKey = dsaKV.Key;
+                    DSAParameters dsaParams = dsaKey.ExportParameters(false);
+
+                    Assert.Equal(Convert.FromBase64String(dsaP), dsaParams.P);
+                    Assert.Equal(Convert.FromBase64String(dsaQ), dsaParams.Q);
+                    Assert.Equal(Convert.FromBase64String(dsaG), dsaParams.G);
+                    Assert.Equal(Convert.FromBase64String(dsaY), dsaParams.Y);
+
+                    // J is an optimization it should either be null or correct value
+                    if (dsaParams.J != null)
+                    {
+                        Assert.Equal(Convert.FromBase64String(dsaJ), dsaParams.J);
+                    }
+
+                    Assert.Equal(Convert.FromBase64String(dsaSeed), dsaParams.Seed);
+
+                    byte[] counter = Convert.FromBase64String(dsaPgenCounter);
+                    Assert.InRange(counter.Length, 1, 4);
+                    int counterVal = 0;
+                    for (int j = 0; j < counter.Length; j++)
+                    {
+                        counterVal <<= 8;
+                        counterVal |= counter[j];
+                    }
+
+                    Assert.Equal(counterVal, dsaParams.Counter);
```

Comment:
I expect this to fail on Linux and/or macOS.  On Linux we don't roundtrip Counter.  Counter and Seed should have the same nullarity, though.  (Counter == null => Seed == null; Seed == null => Counter == null).

Similar to J, if it's present it should be right.